### PR TITLE
Recall, precision, fscore and Hausdorff

### PR DESCRIPTION
**Pull request by myychal**
Added new metrics

### CODE REVIEW

**Comment by szympie** [comment]
Need for change of the form of the metrics emerged with the switch to the ignite framework. Basically, if we would like to use it in ignite, metric should only take predictions and target and return tensor with shape=0 (basically a float, without any []), hence I propose to put metrics in classes with `__call__(prediction, target)` method. 

I think such approach is better even without taking ignite into account:
```python

standard_fscore = FScore(beta=0.5, epsilon=1e-6)
score = standard_fscore(prediction, target)

```

File: brats/metrics.py [hunk 1]
```diff
@@ -1,30 +1,184 @@
...
+class RecallScore:
+    def __init__(self, epsilon: float = 1e-6):
+        """
+        Compute the recall score
+        Args:
+            epsilon: Smooth factor to prevent division by 0.
+        """
+        self.epsilon = epsilon
+
+    def __call__(self, prediction: torch.Tensor,
```
Comment (szympie) [inline brats/metrics.py]: You can do the same thing in all losses and metrics. Extract it to functional form, that is later used in classes

File: brats/losses.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ def __init__(self, epsilon: float = 1e-6):
         """
         super().__init__()
         self.epsilon = epsilon
+        self.dice_score = DiceScoreOneClass()
```
[the remark this answers is not in the document]
Comment (myychal) [reply brats/losses.py]: Extracted to _functional.py_.

File: brats/metrics.py [hunk 3]
```diff
@@ -1,30 +1,184 @@
+import typing
```
[the remark this answers is not in the document]
Comment (myychal) [reply brats/metrics.py]: Corrected

File: brats/metrics.py (hunk 1 above)
Comment (myychal) [reply brats/metrics.py]: Did that, they are all now in _functional.py_.

File: brats/metrics.py [hunk 4]
```diff
@@ -1,30 +1,184 @@
...
+        """
+        precision = self.precision_score(prediction, target)
+        recall = self.recall_score(prediction, target)
+        f_score = (1 + self.beta ** 2) * ((precision * recall) / (
+                self.beta ** 2 * precision + recall + self.epsilon))
+        return torch.mean(f_score)
+
+
+class HausdorffDistance95:
+    def __init__(self, merge_operation=torch.max):
```
[the remark this answers is not in the document]
Comment (myychal) [reply brats/metrics.py]: Added `typing.Callable`

File: brats/utils.py [hunk 5]
```diff
@@ -1,4 +1,31 @@
...
+    """
+    Check whether the input contains only 0s.
+    Args:
+        input: Array to check
+
+    Returns:
+        bool: True if contains only 0s, False otherwise
+
+    """
+    return not np.any(input != 0)
```
Comment (khrynczenko) [inline brats/utils.py]: I think it would be clearer to just write `return np.all(input==0)` :P

Comment (myychal) [reply brats/utils.py]: I an alternative yerba drinker now - such pathetically obvious solutions don't satisfy me anymore
